src/pages/Products/components/Filters/CategoryFilter.jsx

```diff
@@ -1,73 +1,49 @@
-const CategoryFilter = ({ chessPieces, chessBoard, chessBooks, chessClock, filterDispatch }) => {
+const CategoryFilter = ({ category, filterDispatch }) => {
     return (
         <div className="filter-type">
             <h3 className="filter-type-heading">Category</h3>
 
             {/* Chess Pieces Filter */}
             <div className="filter-type-container">
-                <input type="checkbox" id="chess-pieces" checked={chessPieces}
-                    onChange={(event) =>
-                        event.target.checked
-                            ? filterDispatch({
-                                type: "CHESS_PIECES",
-                                payload: { chessPieces: true }
-                            })
-                            : filterDispatch({
-                                type: "CHESS_PIECES",
-                                payload: { chessPieces: false }
-                            })
+                <input type="checkbox" id="chess-pieces" 
+                checked={category.includes("chess-pieces")}
+                onChange={(event) => filterDispatch(
+                    {   type: "CATEGORY_FILTER", 
+                        payload: {categoryType: "chess-pieces", isChecked: event.target.checked}})
```

Comment:
the `isChecked` payload here is not being used in the `filter-reducer`. You can remove it if you are not using it.